src/os_xml/os_xml.c

```diff
@@ -286,6 +286,27 @@ static int _ReadElem(unsigned int parent, OS_XML *_lxml, unsigned int recursion_
         cmp = '\0';
     }
 
+    // consume all the spaces, tabs or new line characters
+    while ((c = xml_getc_fun(_lxml->fp, _lxml)) != cmp) {
+        if (isspace(c)) {
+            continue;
+        } else {
+            break;
+        }
+    }
+
+    // check that the next character is '<'
+    if (c == cmp) {
+        retval = LEOF;
+        xml_error(_lxml, "XMLERR: Empty content.");
+        goto end;
+    }
+    else if (c != _R_CONFS) {
+        xml_error(_lxml, "XMLERR: Malformed XML does not start with '<'");
+        goto end;
+    }
+    _xml_ungetc(_R_CONFS, _lxml);
+
```

Comment:
What if we include to validate the end of the file/string, containing the character '>'. 
Without trailing spaces and returning file position at the beginning of the file when validated by Ok. 

``` c
// Get the current position before checking '>'
long pos = ftell(_lxml->fp);

// validate that the file ends with '>'
fseek(_lxml->fp, -1, SEEK_END);
if (fgetc(_lxml->fp) != '>') {
    xml_error(_lxml, "XMLERR: Malformed XML does not end with '>'");
    fseek(_lxml->fp, pos, SEEK_SET); // Return to the position of '<'
    goto end;
}

// Set the current character back to '<'
fseek(_lxml->fp, pos, SEEK_SET);
c = _R_CONFS;
```

Reply:
The xml parser already validate it.